<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SqlExecutionArea</class>
 <widget class="QWidget" name="SqlExecutionArea">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>365</width>
    <height>482</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="childrenCollapsible">
      <bool>false</bool>
     </property>
     <widget class="SqlTextEdit" name="editEditor"/>
     <widget class="ExtendedTableWidget" name="tableResult">
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
     </widget>
     <widget class="QWidget" name="">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QTextEdit" name="editErrors">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Monospace</family>
           <pointsize>8</pointsize>
          </font>
         </property>
         <property name="acceptDrops">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string>Results of the last executed statements</string>
         </property>
         <property name="whatsThis">
          <string>This field shows the results and status codes of the last executed statements.</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="tabChangesFocus">
          <bool>true</bool>
         </property>
         <property name="undoRedoEnabled">
          <bool>false</bool>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="buttonSave">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="icon">
          <iconset resource="icons/icons.qrc">
           <normaloff>:/icons/save_table</normaloff>:/icons/save_table</iconset>
         </property>
         <property name="popupMode">
          <enum>QToolButton::InstantPopup</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
  <action name="actionExportCsv">
   <property name="text">
    <string>Export to &amp;CSV</string>
   </property>
  </action>
  <action name="actionSaveAsView">
   <property name="text">
    <string>Save as &amp;view</string>
   </property>
   <property name="toolTip">
    <string>Save as view</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SqlTextEdit</class>
   <extends>QTextEdit</extends>
   <header>sqltextedit.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ExtendedTableWidget</class>
   <extends>QTableWidget</extends>
   <header>ExtendedTableWidget.h</header>
   <slots>
    <signal>foreignKeyClicked(QString,QString,QByteArray)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>editEditor</tabstop>
  <tabstop>tableResult</tabstop>
  <tabstop>editErrors</tabstop>
  <tabstop>buttonSave</tabstop>
 </tabstops>
 <resources>
  <include location="icons/icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExportCsv</sender>
   <signal>triggered()</signal>
   <receiver>SqlExecutionArea</receiver>
   <slot>saveAsCsv()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>183</x>
     <y>215</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveAsView</sender>
   <signal>triggered()</signal>
   <receiver>SqlExecutionArea</receiver>
   <slot>saveAsView()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>183</x>
     <y>215</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>saveAsCsv()</slot>
  <slot>saveAsView()</slot>
 </slots>
</ui>
